backend/src/main/scala/cromwell/backend/wdl/WdlFileMapper.scala

```diff
@@ -0,0 +1,28 @@
+package cromwell.backend.wdl
+
+import lenthall.util.TryUtil
+import wdl4s.values.{WdlArray, WdlFile, WdlMap, WdlValue}
+
+import scala.util.{Success, Try}
+
+object WdlFileMapper {
+  def mapWdlFiles(mapper: (WdlFile => WdlFile))(wdlValue: WdlValue): Try[WdlValue] = {
+    wdlValue match {
+      case file: WdlFile => Try(mapper(file))
+      case array: WdlArray =>
+        val mappedArray = array.value map mapWdlFiles(mapper)
+        TryUtil.sequence(mappedArray) map {
+          WdlArray(array.wdlType, _)
+        }
+      case map: WdlMap =>
+        val mappedMap = map.value map {
+          // TODO: TryUtil.sequenceMap doesn't currently handle keys of Try[WdlValue], so for now calling .get here
```

Comment:
[`WdlFile`s as keys](https://github.com/broadinstitute/cromwell/blob/develop/core/src/test/scala/cromwell/util/SampleWdl.scala#L653) were tested within `MapWorkflowSpec`. I had commented out the tested portion as it took some wiring to get to the internally generated workflow id. I've made adjustments and restored the test.